archive.sh

```diff
@@ -191,18 +191,20 @@ repo_packages() {
 
 	touch "$PACKAGES_DIR" "$PKGFLAT"
 
-	(( $REPO_PACKAGES_INDEX )) && repo_packages_index "$PKGFLAT"
+	(( $REPO_PACKAGES_INDEX )) && repo_packages_index "$PACKAGES_DIR" "$PKGFLAT"
 }
 
 # creating a lightweight index (v0) for filenames
 # modification time is used to minimize download, so we don't update the file if equal
 repo_packages_index() {
 	msg 'Updating package index'
-	local INDEX="$1/index.0.xz"
-	local TMPINDEX="$1/.index.0.xz"
+	local source_directory=$1
+	local index_directory=$2
+	local INDEX="$index_directory/index.0.xz"
+	local TMPINDEX="$index_directory/.index.0.xz"
 
 	rm -f "$TMPINDEX"
-	find "$1" -name "*$PKGEXT" -printf '%f\n'|sed 's/.\{'${#PKGEXT}'\}$//'|pacsort|xz -9 > "$TMPINDEX"
+	find "$source_directory" -name "*$PKGEXT" -printf '%f\n'|sed 's/.\{'${#PKGEXT}'\}$//'|sort -u|pacsort|xz -9 > "$TMPINDEX"
```

Comment:
pacsort already does sorting, so you could use uniq after that and avoid needing to process the collation again.

Reply:
Huh, odd. Good point, although IMO this is a pacman-contrib bug and should be reported as such.

Reply:
Actually seems like it might work when both using the .pkg.tar.xz extension and using `pacsort -f`, but I'm unclear why the archivetools remove the `$PKGEXT` anyway. Our tooling does not enforce one, so without listing it in the index we cannot really assert what the proper download name is.